woocommerce/woocommerce-ios
# Initial implementation of Coupon add/editing view

Part of #6488 and #6492

### Description
This is an initial implementation of the add/editing screen of Coupons, with its view model. A super basic screen that could be presented from the ellipsis menu from a coupon detail view. For the moment the only logic implemented is related to the title of the view (that will not be displayed because in the next PRs we will remove the `NavigationView`, which is probably not needed with the current design of the view).

### Testing instructions
1. Launch the app with the new feature flag `couponEditing` enabled.
2. Navigate to a coupon detail.
3. Press the ellipsis button in the navigation bar.
4. You should be able to see the new action "Edit Coupon".
5. Press the "Edit Coupon" action -> The new screen should be presented. 

### Screenshots
<img src="https://user-images.githubusercontent.com/495617/160437276-670b83cb-b837-473b-bea8-4cc4b7c72890.png" width=300 />


---
